### PR TITLE
Improve definition of "frontend plugins"

### PR DESCRIPTION
**Pull request by garvinhicking**
Originating in a slack discussion: https://typo3.slack.com/archives/C079VNXERKN/p1720596676302029

Just a first draft we can maybe improve on, either with follow-ups or here.

### CODE REVIEW

**Review by Alagts — APPROVED** [review]

File: Documentation/ExtensionArchitecture/HowTo/FrontendPlugin/Index.rst [hunk 1]
```diff
@@ -16,8 +29,10 @@ For pure output it is often sufficient to use a
 :ref:`Creating a custom content element <adding-your-own-content-elements>`.
 
 For scenarios with user input and or complicated data operations consider
-using :ref:`Extbase <extbase>`.
+using :ref:`Extbase <extbase>` (specifically
+:ref:`Registration of frontend plugins <t3coreapi:extbase_registration_of_frontend_plugins>`).
 
-It is also possible to create a frontend plugin using Core functionality only.
+It is also possible to create a frontend plugin using Core functionality
+only, using custom content object renderers.
```
[the remark this answers is not in the document]
Comment (fe-hicking) [reply Documentation/ExtensionArchitecture/HowTo/FrontendPlugin/Index.rst:36]: More like generally implementing things like FLUIDTEMPLATE on its own, but yours would also be ok.. i wanted to keep this unspecific to not overcomplicate the intro, and just mention cobj renderers like what fsc/csc implements?

File: Documentation/ExtensionArchitecture/HowTo/FrontendPlugin/Index.rst [hunk 2]
```diff
@@ -8,6 +8,19 @@
...
 
+The term "frontend plugin" describes a part of a TYPO3 extension that is
+handled like a content element (can be inserted like a record/element in
+the TYPO3 backend by editors), which will deliver dynamic output when
+rendered in the frontend. The distinction and boundaries to regular content
+are sometimes not easy to draw, because also "regular" content elements
+are often able to perform dynamic output (for example with TypoScript
+configuration, Fluid data processors or ViewHelpers).
+
+For a long time, TYPO3 provided a "General Plugin" to be selected as a
```
[the remark this answers is not in the document]
Comment (garvinhicking) [reply Documentation/ExtensionArchitecture/HowTo/FrontendPlugin/Index.rst:19]: Adjusted, ok?

File: Documentation/ExtensionArchitecture/HowTo/FrontendPlugin/Index.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (garvinhicking) [reply Documentation/ExtensionArchitecture/HowTo/FrontendPlugin/Index.rst:36]: Agreed!

**Review by linawolf — DISMISSED** [review]

**Review by mbrodala — CHANGES_REQUESTED** [review]

**Review by mbrodala — APPROVED** [review]

**Comment by garvinhicking** [comment]
Created manual backports and a follow-up. Seems my last commit made in the GitHub UI here didn't make it through or was kiled in a rebase. -.-

**Pull request merged**